zipline/pipeline/pipeline.py

```diff
@@ -6,8 +7,10 @@
     optional,
 )
 
-from .graph import ExecutionPlan, TermGraph
+from .domain import Domain, infer_domain
+from .graph import ExecutionPlan, TermGraph, _SCREEN_NAME
 from .filters import Filter
+from .sentinels import NotSpecified, NotSpecifiedType
```

Comment:
the pattern Python uses for None and Pandas uses for NaT is just to use `type(sentinel)` instead of importing the type directly.

Reply:
I used `NotSpecifiedType` because that was the pattern we already had in a bunch of places. I could see an argument for updating this, but I'd rather do that in a separate atomic PR instead of rolling more miscellaneous changes into this.